### PR TITLE
Add default issue metadata in testcase upload.

### LINKED ISSUE #1191: Add default testcase metadata (components, hotlists) for testcase uploads.

### PR DESCRIPTION
**Pull request by inferno-chromium**
Fixes #1191

### CODE REVIEW

File: src/python/bot/tasks/analyze_task.py [hunk 1]
```diff
@@ -36,6 +38,36 @@
...
+  """Adds the default issue metadata (e.g. components, labels) to testcase."""
+  default_metadata = engine_common.get_all_issue_metadata_for_testcase(testcase)
+  if not default_metadata:
+    return
+
+  testcase_metadata = testcase.get_metadata()
+  for key, default_value in six.iteritems(default_metadata):
+    # Initialize new value list to the value from default issue metadata.
+    # This needs to be appended first since component preference should be given
+    # to existing testcase metadata value.
```
Comment (oliverchang) [inline src/python/bot/tasks/analyze_task.py]: this comment seems a bit confusing to me. how about rewording to something like:

"Add the default issue metadata first. This gives preference to uploader specified issue metadata."

File: src/python/bot/tasks/analyze_task.py [hunk 2]
```diff
@@ -36,6 +38,36 @@
...
+    new_value_list = utils.parse_delimited(default_value, delimiter=',')
+
+    # Append existing testcase metadata value to end (for preference).
```
Comment (oliverchang) [inline src/python/bot/tasks/analyze_task.py]: nit: s/existing/uploader specified/

File: src/python/bot/tasks/analyze_task.py [hunk 3]
```diff
@@ -36,6 +38,36 @@
...
+    current_value = testcase_metadata.get(key)
```
Comment (oliverchang) [inline src/python/bot/tasks/analyze_task.py]: nit: s/current_value/uploader_value/

File: src/python/bot/tasks/analyze_task.py [hunk 4]
```diff
@@ -36,6 +38,36 @@
...
+    if current_value:
```
Comment (oliverchang) [inline src/python/bot/tasks/analyze_task.py]: nit: maybe we can avoid this indentation by doing:

```python
uploader_value = testcase_metadata.get(key, '')
uploader_value_list = utils.parse_delimited(uploader_value, delimiter=',')
for value in uploader_value_list:
 ...
```

File: src/python/bot/tasks/analyze_task.py (hunk 1 above)
Comment (inferno-chromium) [reply src/python/bot/tasks/analyze_task.py]: Done!

File: src/python/bot/tasks/analyze_task.py (hunk 2 above)
Comment (inferno-chromium) [reply src/python/bot/tasks/analyze_task.py]: Done!

File: src/python/bot/tasks/analyze_task.py (hunk 3 above)
Comment (inferno-chromium) [reply src/python/bot/tasks/analyze_task.py]: Done!

File: src/python/bot/tasks/analyze_task.py (hunk 4 above)
Comment (inferno-chromium) [reply src/python/bot/tasks/analyze_task.py]: Done!
